lib/connections/service.go

```diff
@@ -153,12 +152,14 @@ next:
 			continue
 		}
 
-		hello, err := exchangeHello(c, s.model.GetHello(remoteID))
+		c.SetDeadline(time.Now().Add(20 * time.Second))
+		hello, err := protocol.ExchangeHello(c, s.model.GetHello(remoteID))
 		if err != nil {
 			l.Infof("Failed to exchange Hello messages with %s (%s): %s", remoteID, c.RemoteAddr(), err)
```

Comment:
Nah... we print info level message for every other occurrence of something interesting in the connection loop so I think we should do that here too.